weaviate/collections/batch/client.py

```diff
@@ -105,6 +113,73 @@ def __enter__(self) -> _BatchClient:
         )
         return self._current_batch
 
+    def __create_batch_and_reset(self) -> _ContextManagerWrapper[_BatchClient]:
+        self._batch_data = _BatchDataWrapper()  # clear old data
+        return _ContextManagerWrapper(
+            _BatchClient(
+                connection=self._connection,
+                consistency_level=self._consistency_level,
+                results=self._batch_data,
+                batch_mode=self._batch_mode,
+            )
+        )
+
+    def dynamic(
+        self, consistency_level: Optional[ConsistencyLevel] = None
+    ) -> _ContextManagerWrapper[_BatchClient]:
+        """Configure dynamic batching.
+
+        When you exit the context manager, the final batch will be sent automatically.
+
+        Arguments:
+            `consistency_level`
+                The consistency level to be used to send batches. If not provided, the default value is `None`.
+        """
+        self._batch_mode: _BatchMode = _DynamicBatching()
+        self._consistency_level = consistency_level
+        return self.__create_batch_and_reset()
+
+    def fixed_size(
+        self,
+        batch_size: int = 100,
+        concurrent_requests: int = 2,
+        consistency_level: Optional[ConsistencyLevel] = None,
+    ) -> _ContextManagerWrapper[_BatchClient]:
+        """Configure fixed size batches. Note that the default is dynamic batching.
+
+        When you exit the context manager, the final batch will be sent automatically.
+
+        Arguments:
+            `batch_size`
+                The number of objects/references to be sent in one batch. If not provided, the default value is 100.
+            `concurrent_requests`
+                The number of concurrent requests when sending batches. This controls the number of concurrent requests
+                made to Weaviate and not the speed of batch creation within Python.
+            `consistency_level`
+                The consistency level to be used to send batches. If not provided, the default value is `None`.
+
+        """
+        self._batch_mode = _FixedSizeBatching(batch_size, concurrent_requests)
+        self._consistency_level = consistency_level
+        return self.__create_batch_and_reset()
+
+    def rate_limit(
+        self, requests_per_minute: int, consistency_level: Optional[ConsistencyLevel] = None
```

Comment:
Would an `api_` prefix here make it clearer which request is in question? Might help with the potential confusion JP raised this morning